graphqlapi/src/main/java/io/stargate/graphql/web/resources/schemafirst/NamespaceResource.java

```diff
@@ -59,7 +51,7 @@ public void get(
       @Context HttpServletRequest httpRequest,
       @Suspended AsyncResponse asyncResponse) {
 
-    GraphQL graphql = getGraphl(namespace, httpRequest, asyncResponse);
+    GraphQL graphql = getGraphql(namespace, httpRequest, asyncResponse, graphqlCache);
```

Comment:
It is not clear to me how it should work (and how it works now).
1. Is the `NamespaceResource` serving the schema based on version? If yes, why it needs to accept the query, variables, and other query parameters? Shouldn't it just return the schema for a given version?
A similar pattern is for `AdminResource` get method. Shouldn't we just return the `SchemaSource` from the new endpoint?

Reply:
There are two aspects of the schema:
* the "raw" GraphQL string that the user passed to `deploySchema`. This is what we store in the `graphql_schema` table.
* the "runtime" `GraphQL` object that we use to execute queries. We build it by processing the raw schema, adding any generated types, binding the fetchers, etc. (see `SchemaProcessor`).

In this ticket, the goal is to expose the raw schema (so that users can download a previous version of their schema, modify it locally and re-deploy it).

> Shouldn't we just return the `SchemaSource` from the new endpoint?

Correct. `NamespaceResource` handles user queries on an already deployed schema, so it deals with the "runtime" schema. It's not the right place to look, instead the new REST resource should be more similar to `SingleSchemaFetcher`: call `SchemaSourceDao.get()` for the provided namespace and version, and then serve the `contents` field. You'll have to register a DAO instance in the Jersey environment in order to inject it in the resource, see `DropwizardServer`.
`SchemaFirstCache` builds its own DAO instance, so we might want to refactor to reuse a single one instead.

I haven't done a great job with naming. Maybe the `schema*` admin queries should have some notion of "raw" or "source" in the name, I guess that might be confusing for users too. Also I don't have a good name for the "non-admin" GraphQL schema, i.e. what gets deployed to a namespace (your question shows that `NamespaceResource` was a poor choice). I thought of "runtime", but that's not the best description because admin queries execute at runtime too. If you have suggestions I'd like to hear them.

Reply:
The `SchemaSourceDao` accepts `DataStore` as an argument. We could remove it from the constructor, and add pass it as an argument to every method that needs it. However, as I see the `SchemaSourceDao` is stateless (besides the usage of the `dataStore`) so I think we can create a new instance of it wherever it's needed. Also, there are some code paths that do not have `DataStore` instance (i.e when `SchemaSourceDao` is called from *Resource),
wdyt?
Regarding naming, maybe:
- `RawSchema` and `ProcessedSchema`?

Reply:
> However, as I see the `SchemaSourceDao` is stateless (besides the usage of the `dataStore`) so I think we can create a new instance of it wherever it's needed.

Thew way you did it in `DropwizardServer` looks good to me.